sphinx/application.py

```diff
@@ -321,16 +322,20 @@ def _init_env(self, freshenv):
                     logger.info('failed: %s', err)
                 self._init_env(freshenv=True)
 
-    def _init_builder(self, buildername):
+    def _init_builders(self, buildernames):
         # type: (unicode) -> None
-        if buildername is None:
+        if not len(buildernames):
             print('No builder selected, using default: html', file=self._status)
-            buildername = 'html'
-        if buildername not in self.builderclasses:
-            raise SphinxError('Builder name %s not registered' % buildername)
+            buildernames.append('html')
 
-        builderclass = self.builderclasses[buildername]
-        self.builder = builderclass(self)
+        for builder in buildernames:
+            if builder not in self.builderclasses:
+                raise SphinxError('Builder name %s not registered' % builder)
+
+            builderclass = self.builderclasses[builder]
+            self.builders[builder] = builderclass(self)
+
+        self.builder = self.builders[buildernames[0]]
         self.emit('builder-inited')
```

Comment:
I'm inclined to say that, semantically, this should pass the initialized builder, but it looked like this would require an update for every subscriber to this event.